Project2015To2017/Reading/SolutionReader.cs

```diff
@@ -11,35 +11,74 @@ public sealed class SolutionReader
 	{
 		public static readonly SolutionReader Instance = new SolutionReader();
 
+		// An example of a project line looks like this:
+		//  Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ClassLibrary1", "ClassLibrary1\ClassLibrary1.csproj", "{05A5AD00-71B5-4612-AF2F-9EA9121C4111}"
+		private static readonly Lazy<Regex> crackProjectLine = new Lazy<Regex>(
+			() => new Regex
+			(
+				"^" // Beginning of line
+				+ "Project\\(\"(?<PROJECTTYPEGUID>.*)\"\\)"
+				+ "\\s*=\\s*" // Any amount of whitespace plus "=" plus any amount of whitespace
+				+ "\"(?<PROJECTNAME>.*)\""
+				+ "\\s*,\\s*" // Any amount of whitespace plus "," plus any amount of whitespace
+				+ "\"(?<RELATIVEPATH>.*)\""
+				+ "\\s*,\\s*" // Any amount of whitespace plus "," plus any amount of whitespace
+				+ "\"(?<PROJECTGUID>.*)\""
+				+ "$", // End-of-line
+				RegexOptions.Compiled
+			)
+		);
+
+		private const string vbProjectGuid = "{F184B08F-C81C-45F6-A57F-5ABD9991F28F}";
+		private const string csProjectGuid = "{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}";
```

Comment:
are you sure this covers all generations of Visual Studio and things like test projects? What about WPF? I had a feeling that there were more possible values it could take.

Reply:
These are not the ProjectTypeGuids from project file. These are specific solution Guids, that cover all managed languages supported by latest MSBuild but J# and C++/CLI. This was taken directly from MSBuild sources.